b2share/modules/records/serializers/schemas/datacite.py

```diff
@@ -119,6 +119,17 @@ def get_contributors(self, obj):
                 for c in obj['metadata'].get('contributors', [])]
 
     def get_descriptions(self, obj):
-        return [{'description': d['description'],
+        items = []
+        for d in obj['metadata'].get('descriptions', []):
+            d = description_to_datacite_v3(d)
```

Comment:
What about changing that into:
```python
datacite_v3_description_types = set([<ALLOWED TYPES>])
def get_descriptions(self, obj):
    return [
        {
            'description': description['description'],
            'descriptionType': description['description_type']
                if description['description_type'] in datacite_v3_description_types else 'Other',
        }
        for description in obj['metadata'].get('descriptions', [])
    ]
```
It would be more efficient as there are less functions called and no `append` call. You can see [this page](https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Loops) for more details.
The suggested implementation also allows to have more custom types without having to ever change this function.